feat: Add workaround to hide cursor in screensaver

alexperreault opened the pull request:
This PR is a hack to hide the cursor when the screensaver is on, solving #492

It works by using ```hyprctl keyword```, which allows changing the config from the cli temporarily (until next hyprland relaunch, which goes back to default). It updates the cursor:inactive_timeout setting to 1 second when starting the screensaver, then disables the timeout when quitting the screensaver.

This has the potential downside of temporarily overriding the config of a user using the cursor:inactive_timeout setting, until next hyprland relaunch

dhh commented:
What would happen if the screensaver is interrupted by CTRL+C or the like? Couldn't the config get stuck? Do we need to restore it in a trap or something?

alexperreault commented:
You are right, the config gets stuck.

I'm not familiar with traps, i'll read a bit and figure something out

alexperreault commented:
Fixed the behavior with multiple monitors / multiple instances of the screensaver.

Previously, the original timeout config was saved when each individual screensaver was launched. The first screensaver had the correct value, but every following one saved the overwritten 1 value, meaning if a screensaver other than the first one launched was responsible to close the screensavers, the incorrect timeout value was applied.

Also, fixed that a ctrl+c on one screensaver closes all screensaver using trap. 

Tested successfully on one and two monitors, closing with ctrl+c or any key.

alexperreault commented:
Coming in next Hyprland version is an option to make the cursor invisible instead of changing the timeout value. Closing for now
